sources/api/thar-be-settings/src/service.rs

```diff
@@ -4,36 +4,49 @@ use std::collections::HashSet;
 use std::path::Path;
 use std::process;
 
-use itertools::join;
+use itertools::{join, Itertools};
 
 use crate::{error, Result};
 
-/// Wrapper for the multiple functions needed to go from
-/// a list of changed settings to a Services map
-#[allow(clippy::implicit_hasher)]
-pub async fn get_affected_services<P>(
+//returns changed settings corresponding to the services
+pub async fn get_affected_services_settings_map<P>(
     socket_path: P,
     settings_limit: Option<HashSet<String>>,
-) -> Result<model::Services>
+) -> Result<Option<HashMap<String, Vec<String>>>>
 where
     P: AsRef<Path>,
 {
-    let service_limit = if let Some(settings_limit) = settings_limit {
+    let settings_limit = if let Some(settings_limit) = settings_limit {
         let setting_to_service_map =
             get_affected_service_map(socket_path.as_ref(), settings_limit).await?;
-        if setting_to_service_map.is_empty() {
-            return Ok(HashMap::new());
-        }
-
-        let service_names = get_affected_service_names(setting_to_service_map);
-        Some(service_names)
+        //if the settings do not affect any services, empty map is returned
+        let service_settings_map = get_service_settings_map(&setting_to_service_map);
+        info!(
+            "Changed settings for each service {:?}",
+            &service_settings_map
+        );
+        Some(service_settings_map)
     } else {
         // No limit, we want all services.
         None
     };
 
-    let services = get_service_metadata(socket_path.as_ref(), service_limit).await?;
+    Ok(settings_limit)
+}
 
+pub async fn get_affected_services<P>(
```

Comment:
Can you try moving this function above `get_affected_services_settings_map` so that it starts around line 12 like it did before? I think it might help Git produce a better diff.

Reply:
Yeah sure